### PR TITLE
Refactor/the node job - part 1

### PR DESCRIPTION
**Pull request by Aprogiena**
This moves most of NBitcoin.Protocol to full node code

- lots of files are moved
- large amount of files are changed due to new dependencies
- very minor polishing
- there are NO changes in logic or design
- some NBitcoin tests are commented out, can be ported later

### CODE REVIEW

**Comment by Aprogiena** [comment]
ok, no problem to leave your files where they are, they can be moved later if needed

there still is NBitcoin Protocol folder, because many other components of NBitcoin rely on those files
so i tried to extract the minimum from it so that I can move the Node file to FN. so that is expected and moving the rest of the protocol is bigger task again, no need for that at the moment

**Comment by Aprogiena** [comment]
now that the jeremy's monster is merged and conflicts resolved here, it is time for this one ...

File: src/Stratis.Bitcoin.IntegrationTests/RPC/RPCTests.cs [hunk 1]
```diff
@@ -1,9 +1,9 @@
-﻿using System;
-using System.IO;
 using NBitcoin;
-using NBitcoin.Protocol;
 using NBitcoin.RPC;
```
Comment (bokobza) [inline src/Stratis.Bitcoin.IntegrationTests/RPC/RPCTests.cs]: namespaces are the wrong way around.

File: src/Stratis.Bitcoin/Base/ChainHeadersBehavior.cs [hunk 2]
```diff
@@ -1,14 +1,14 @@
-﻿#if !NOSOCKET
-
-using System;
-using System.Linq;
-using System.Threading;
 using Microsoft.Extensions.Logging;
```
Comment (bokobza) [inline src/Stratis.Bitcoin/Base/ChainHeadersBehavior.cs]: please sort namespaces.

**Comment by bokobza** [comment]
> there are changes in logic or design

where?

**Comment by Aprogiena** [comment]
@bokobza sorry, fixed, NO changes :)

File: src/Stratis.Bitcoin/Base/ChainHeadersBehavior.cs (hunk 2 above)
Comment (Aprogiena) [reply src/Stratis.Bitcoin/Base/ChainHeadersBehavior.cs]: tx, will fix

File: src/Stratis.Bitcoin.IntegrationTests/RPC/RPCTests.cs (hunk 1 above)
Comment (Aprogiena) [reply src/Stratis.Bitcoin.IntegrationTests/RPC/RPCTests.cs]: tx, will fix

**Pull request merged**